Jasmine/Common/Models/TextGrid.swift

```diff
@@ -0,0 +1,61 @@
+struct TextGrid {
+    /// The actual grid.
+    private var grid: [[String?]]
+
+    /// Initializes a CharacterGrid, given the initial grid.
+    ///
+    /// - Parameters:
+    ///   - initialGrid: the initial grid. The CharacterGrid will read from this grid.
+    ///   - randomized: pass in true if the resulting grid wants to be randomized from the initial grid.
+    init(fromInitialGrid initialGrid: [[String]], randomized: Bool) {
```

Comment:
Nah, it's just an easy change haha. The current 2d array representation has type `String?`.